pyserini/encode/_base.py

```diff
@@ -88,6 +88,25 @@ def __iter__(self):
                 to_yield[key] = self.all_info[key][idx: min(idx + self.batch_size, end_idx)]
             yield to_yield
 
+    def _parse_fields_from_contents(self, contents):
```

Comment:
adding a separate function to process the contents, deciding if we should remove the trailing \n. 
As it seems CACM has a trailing \n for all documents